src/kernel/faction.c

```diff
@@ -333,7 +333,8 @@ variant read_faction_reference(struct storage * store)
 
 void write_faction_reference(const faction * f, struct storage *store)
 {
-    WRITE_INT(store, (f && f->_alive) ? f->no : 0);
```

Comment:
this is where it was all going wrong. two different situations (no faction and dead faction) represented by the same value.